readthedocs/search/tests/test_views.py

```diff
@@ -0,0 +1,147 @@
+import pytest
+from django.core.management import call_command
+from django.core.urlresolvers import reverse
+from django_dynamic_fixture import G
+from pyquery import PyQuery as pq
+
+from readthedocs.builds.constants import LATEST
+from readthedocs.builds.models import Version
+from readthedocs.search.tests.utils import get_search_query
+
+
+@pytest.mark.django_db
+@pytest.mark.search
+class TestElasticSearch(object):
+    url = reverse('search')
+
+    def _reindex_elasticsearch(self, es_index):
+        call_command('reindex_elasticsearch')
+        es_index.refresh_index()
+
+    def _get_search_result(self, url, client, search_params):
+        resp = client.get(url, search_params)
+        assert resp.status_code == 200
+
+        page = pq(resp.content)
+        result = page.find('.module-list-wrapper .module-item-title')
+        return result, page
+
+    @pytest.fixture(autouse=True)
+    def elastic_index(self, mock_parse_json, all_projects, es_index):
+        self._reindex_elasticsearch(es_index=es_index)
+
+    def test_search_by_project_name(self, client, project):
+        result, _ = self._get_search_result(url=self.url, client=client,
+                                            search_params={'q': project.name})
+
+        assert project.name.encode('utf-8') in result.text().encode('utf-8')
+
+    @pytest.mark.parametrize('data_type', ['content', 'headers', 'title'])
+    @pytest.mark.parametrize('page_num', [0, 1])
+    def test_search_by_file_content(self, client, project, data_type, page_num):
+        query = get_search_query(project_slug=project.slug, page_num=page_num,
+                                 data_type=data_type)
+
+        versions = project.versions.all()
+        # There should be only one version of the project
+        assert len(versions) == 1
```

Comment:
Not sure this should live in a helper method, so it doesn't seem like it will always be true.